### PR TITLE
feat: migrate less to token for Tooltip

### PR DESCRIPTION
**Pull request by guan404ming**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 💡 Background and solution

ref #41884 

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |      migrate less to token for Tooltip     |
| 🇨🇳 Chinese |     Tooltip 组件less变量迁移      |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at 29ed060</samp>

Refactored the tooltip style code and updated the documentation for migrating less variables to component tokens. This improves the consistency and customizability of the tooltip component.

### <samp>🤖 Generated by Copilot at 29ed060</samp>

*  Extend `ComponentToken` interface to include common tooltip properties and simplify `TooltipToken` interface ([link](https://github.com/ant-design/ant-design/pull/42046/files?diff=unified&w=0#diff-6cf5aa7d06726f3671a27bb4ba86ebe2d1c8d05e0ff53fab95d09b4c238132e0L10-R22))
*  Refactor `genTooltipStyle` value to use `token` parameter directly instead of destructuring it ([link](https://github.com/ant-design/ant-design/pull/42046/files?diff=unified&w=0#diff-6cf5aa7d06726f3671a27bb4ba86ebe2d1c8d05e0ff53fab95d09b4c238132e0L41-R32), [link](https://github.com/ant-design/ant-design/pull/42046/files?diff=unified&w=0#diff-6cf5aa7d06726f3671a27bb4ba86ebe2d1c8d05e0ff53fab95d09b4c238132e0L50-R51), [link](https://github.com/ant-design/ant-design/pull/42046/files?diff=unified&w=0#diff-6cf5aa7d06726f3671a27bb4ba86ebe2d1c8d05e0ff53fab95d09b4c238132e0L76-R64), [link](https://github.com/ant-design/ant-design/pull/42046/files?diff=unified&w=0#diff-6cf5aa7d06726f3671a27bb4ba86ebe2d1c8d05e0ff53fab95d09b4c238132e0L106-R98))
*  Move default tooltip properties from `mergeToken` call to second argument of `useComponentToken` hook in `components/tooltip/style/index.ts` ([link](https://github.com/ant-design/ant-design/pull/42046/files?diff=unified&w=0#diff-6cf5aa7d06726f3671a27bb4ba86ebe2d1c8d05e0ff53fab95d09b4c238132e0L136-R137))
*  Update documentation for migrating less variables to component tokens to include tooltip component in `docs/react/migrate-less-variables.en-US.md` and `docs/react/migrate-less-variables.zh-CN.md` ([link](https://github.com/ant-design/ant-design/pull/42046/files?diff=unified&w=0#diff-999566baa64628c49813370e7cb13c3c8f5e0c99ef8fe9cec7c5052985c10d9aR39-R51), [link](https://github.com/ant-design/ant-design/pull/42046/files?diff=unified&w=0#diff-dff3e0a0a5fae848621e55da4e2adaf0ec6c57d5e29d3f601799c8898c280739R39-R51))

### CODE REVIEW

File: components/tooltip/style/index.ts [hunk 1]
```diff
@@ -7,60 +7,48 @@ import { genComponentStyleHook, mergeToken, genPresetColor } from '../../theme/i
...
-  } = token;
+  const { componentCls } = token;
 
   return [
     {
       [componentCls]: {
         ...resetComponent(token),
         position: 'absolute',
-        zIndex: zIndexPopup,
+        zIndex: token.zIndexPopup,
```
Comment (MadCcc) [inline components/tooltip/style/index.ts:29]: 尽量使用解构语法，体积会略小一丢丢

File: components/tooltip/style/index.ts [hunk 2]
```diff
@@ -7,60 +7,48 @@ import { genComponentStyleHook, mergeToken, genPresetColor } from '../../theme/i
...
         width: 'max-content',
-        maxWidth: tooltipMaxWidth,
+        maxWidth: token.tooltipMaxWidth,
         visibility: 'visible',
         '&-hidden': {
           display: 'none',
         },
 
-        '--antd-arrow-background-color': tooltipBg,
+        '--antd-arrow-background-color': token.tooltipBg,
```
Comment (MadCcc) [inline components/tooltip/style/index.ts:38]: 这个可以对应上 arrowBg

File: docs/react/migrate-less-variables.zh-CN.md [hunk 3]
```diff
@@ -36,3 +36,16 @@ title: less 变量迁移 Component Token
...
+
+## 文字提示
+
+<!-- prettier-ignore -->
+| less 变量 | Component Token | 备注 |
+| --- | --- | --- |
+| `@tooltip-max-width` | `tooltipMaxWidth` | - |
+| `@tooltip-color` | `tooltipColor` | - |
+| `@tooltip-bg` | `tooltipBg` | - |
+| `@tooltip-arrow-width` | - | 由于样式变化已废弃 |
```
Comment (MadCcc) [inline docs/react/migrate-less-variables.zh-CN.md:48]: 这个可以对应 `sizePopupArrow`

File: docs/react/migrate-less-variables.zh-CN.md [hunk 4]
```diff
@@ -36,3 +36,16 @@ title: less 变量迁移 Component Token
 | `@modal-confirm-body-padding` | - | 由于样式变化已废弃 |
 | `@modal-confirm-title-font-size` | `modalHeaderTitleFontSize` | - |
 | `@modal-border-radius` | `borderRadiusLG` | - |
+
+## 文字提示
```
Comment (MadCcc) [inline docs/react/migrate-less-variables.zh-CN.md:40]:
```suggestion
## Tooltip 文字提示
```

File: components/tooltip/style/index.ts (hunk 2 above)
Comment (guan404ming) [reply components/tooltip/style/index.ts:38]: 不太確定這個改的對不對，還得麻煩大佬再看看，感謝

**Comment by kiner-tang** [comment]
@guan404ming 记得添加一下 debug demo 哟！可以参考一下这个 pr :
- https://github.com/ant-design/ant-design/pull/42342

**Comment by guan404ming** [comment]
> @guan404ming 记得添加一下 debug demo 哟！可以参考一下这个 pr :
> 
> * [✨ feat: Breadcrumb Component Token #42342](https://github.com/ant-design/ant-design/pull/42342)

OK! I will work on it~

File: docs/react/migrate-less-variables.zh-CN.md [hunk 5]
```diff
@@ -368,7 +368,18 @@ title: Less 变量迁移 Design Token
...
 
-<!-- ### Tooltip 文字提示 -->
+### Tooltip 文字提示
+
+<!-- prettier-ignore -->
+| less 变量 | Component Token | 备注 |
+| --- | --- | --- |
+| `@tooltip-max-width` | `tooltipMaxWidth` | - |
+| `@tooltip-color` | `tooltipColor` | - |
+| `@tooltip-bg` | `tooltipBg` | - |
```
Comment (MadCcc) [inline docs/react/migrate-less-variables.zh-CN.md:378]:
```suggestion
| `@tooltip-bg` | `colorBgSpotlight` | 全局 token |
```

File: docs/react/migrate-less-variables.zh-CN.md [hunk 6]
```diff
@@ -368,7 +368,18 @@ title: Less 变量迁移 Design Token
...
+| `@tooltip-arrow-width` | - | 由于样式变化已废弃 |
```
Comment (MadCcc) [inline docs/react/migrate-less-variables.zh-CN.md:379]:
```suggestion
| `@tooltip-arrow-width` | `sizePopupArrow` | Global Token |
```

File: docs/react/migrate-less-variables.zh-CN.md [hunk 7]
```diff
@@ -368,7 +368,18 @@ title: Less 变量迁移 Design Token
...
+| `@tooltip-distance` | - | 由于样式变化已废弃 |
```
Comment (MadCcc) [inline docs/react/migrate-less-variables.zh-CN.md:380]:
```suggestion
| `@tooltip-distance` | `marginXXS` | 组件 Token |
```

File: docs/react/migrate-less-variables.zh-CN.md [hunk 8]
```diff
@@ -368,7 +368,18 @@ title: Less 变量迁移 Design Token
...
+| `@tooltip-arrow-color` | - | 由于样式变化已废弃 |
```
Comment (MadCcc) [inline docs/react/migrate-less-variables.zh-CN.md:381]:
```suggestion
| `@tooltip-arrow-color` | - | 同 `@tooltip-bg`，已废弃 |
```

File: docs/react/migrate-less-variables.zh-CN.md [hunk 9]
```diff
@@ -368,7 +368,18 @@ title: Less 变量迁移 Design Token
...
+| `@tooltip-border-radius` | `tooltipBorderRadius` | - |
```
Comment (MadCcc) [inline docs/react/migrate-less-variables.zh-CN.md:382]:
```suggestion
| `@tooltip-border-radius` | `borderRadius` | 全局 Token |
```

File: docs/react/migrate-less-variables.zh-CN.md [hunk 10]
```diff
@@ -368,7 +368,18 @@ title: Less 变量迁移 Design Token
...
 <!-- ### TimePicker 时间选择 -->
 
-<!-- ### Tooltip 文字提示 -->
+### Tooltip 文字提示
+
+<!-- prettier-ignore -->
+| less 变量 | Component Token | 备注 |
+| --- | --- | --- |
+| `@tooltip-max-width` | `tooltipMaxWidth` | - |
+| `@tooltip-color` | `tooltipColor` | - |
```
Comment (MadCcc) [inline docs/react/migrate-less-variables.zh-CN.md:377]:
```suggestion
| `@tooltip-color` | `colorTextLightSolid` | 全局 Token |
```

File: docs/react/migrate-less-variables.zh-CN.md [hunk 11]
```diff
@@ -368,7 +368,18 @@ title: Less 变量迁移 Design Token
 
 <!-- ### TimePicker 时间选择 -->
 
-<!-- ### Tooltip 文字提示 -->
+### Tooltip 文字提示
+
+<!-- prettier-ignore -->
+| less 变量 | Component Token | 备注 |
+| --- | --- | --- |
+| `@tooltip-max-width` | `tooltipMaxWidth` | - |
```
Comment (MadCcc) [inline docs/react/migrate-less-variables.zh-CN.md:376]:
```suggestion
| `@tooltip-max-width` | - | 可由 `className` 或 `style` 直接修改 |
```
这个就废弃掉了哈

**Comment by MadCcc** [comment]
这样看 Tooltip 就没有其他组件 token 了，可以不用留 Demo 了

File: components/tooltip/style/index.ts [hunk 12]
```diff
@@ -5,63 +5,50 @@ import type { FullToken, GenerateStyle, UseComponentStyleResult } from '../../th
 import { genComponentStyleHook, genPresetColor, mergeToken } from '../../theme/internal';
 
 export interface ComponentToken {
-  zIndexPopup: number;
```
Comment (MadCcc) [inline components/tooltip/style/index.ts:8]: 这个还是留着哈

File: components/tooltip/style/index.ts [hunk 13]
```diff
@@ -135,22 +122,19 @@ export default (prefixCls: string, injectStyle: boolean): UseComponentStyleResul
...
     },
-    ({ zIndexPopupBase, colorBgSpotlight }) => ({
-      zIndexPopup: zIndexPopupBase + 70,
-      colorBgDefault: colorBgSpotlight,
+    (token) => ({
+      colorBgDefault: token.colorBgSpotlight,
+      tooltipMaxWidth: 250,
+      tooltipColor: token.colorTextLightSolid,
+      tooltipBorderRadius: token.borderRadius,
+      tooltipBg: token.colorBgSpotlight,
```
Comment (MadCcc) [inline components/tooltip/style/index.ts:137]: 这边都还原回去吧，应该只用留文档就可以了

Comment (guan404ming) [reply components/tooltip/style/index.ts]: 已經都還原回去了～感謝大佬

Comment (guan404ming) [reply components/tooltip/style/index.ts]: @MadCcc 再麻煩大佬 CR 了～

**Review by MadCcc — APPROVED** [review]

**Pull request merged**